File: Ejer7.java
import java.util.Calendar;
import java.util.GregorianCalendar;
import java.util.Scanner;
import java.util.TimeZone;

public class Ejer7 {
	public static void main(String[] args) {
		Scanner leer = new Scanner(System.in);
		String datos [] = leer.nextLine().split(" ");
		String fecha [] = datos[0].split("/");
		int x=diaSemana(Integer.parseInt(fecha[0]), Integer.parseInt(fecha[1]), Integer.parseInt(fecha[2]));
		if(Integer.parseInt(datos[1])%2==0 )
		{
			if(x==3 || x==5)
				System.out.println("SI");
			else
				System.out.println("NO");
		}
		else
		{
			if(x==2 || x==4 || x==6)
				System.out.println("SI");
			else
				System.out.println("NO");
		}
		
		
	}
	public static int diaSemana (int dia, int mes, int ano)
    {
        TimeZone timezone = TimeZone.getDefault();
        Calendar calendar = new GregorianCalendar(timezone);
        calendar.set(ano, mes-1, dia);
        int nD=calendar.get(Calendar.DAY_OF_WEEK);
        return nD;
    }

}

File: Ejer7.py
from datetime import datetime 
datos=input().split()
fecha=datos[0].split("/")
y=datetime(int(fecha[2]),int(fecha[1]),int(fecha[0]));
x=datetime.weekday(y)
if(int(datos[1])%2==0):
	if(x==1 or x==3):
		print("SI")
	else:
		print("NO")
else:
	if(x==0 or x==2 or x==4):
		print("SI")
	else:
		print("NO")
